GitCommands/SynchronizedProcessReader.cs

```diff
@@ -11,23 +13,21 @@ public class SynchronizedProcessReader
         public byte[] Output { get; private set; }
         public byte[] Error { get; private set; }
 
-        private readonly Thread _stdOutputLoaderThread;
-        private readonly Thread _stdErrLoaderThread;
+        private readonly JoinableTask _stdOutputLoaderThread;
+        private readonly JoinableTask _stdErrLoaderThread;
 
         public SynchronizedProcessReader(Process process)
         {
             Process = process;
-            _stdOutputLoaderThread = new Thread(_ => Output = ReadByte(Process.StandardOutput.BaseStream));
-            _stdOutputLoaderThread.Start();
-            _stdErrLoaderThread = new Thread(_ => Error = ReadByte(Process.StandardError.BaseStream));
-            _stdErrLoaderThread.Start();
+            _stdOutputLoaderThread = ThreadHelper.JoinableTaskFactory.RunAsync(async () => Output = await ReadByteAsync(Process.StandardOutput.BaseStream).ConfigureAwait(false));
```

Comment:
Why `ConfigureAwait(false)` is needed here?

Reply:
➡️ It's more that `ConfigureAwait(true)` is certainly *not* needed here, since nothing directly follows the operation. I was able to avoid the explicit ConfigureAwait in most cases by passing through, but this lambda is marked `async` with an `await` used so the result of the asynchronous evaluation can be assigned to a property.